client/src/pages/Campaign.js

```diff
@@ -122,26 +170,29 @@ const Campaign = (props) => {
      */
   }
 
-  const addStep = (step) => {
-    console.log('Add: ' + JSON.stringify(step));
-    // update UI
-    // fill in template name, sent, and replied
-    step.templateName = findTemplate(step.templateId).name;
-    step.sent = 0;
-    step.replied = 0;
-    // append step to campaign
-    campaign.steps.push(step);
-    setCampaign(campaign);
-
-    /** 
-     * TODO:
-     * update server
-     * create new step with campaign_id=campaign.id
-     * and template_id=step.templateId
-     * 
-     * add step id after request 
-     * (can't update without id, but can't know id until request)
-     */
+  const addNewStep = async () => {
+    const id = campaign.id;
+    const data = {
+      id : templateId
+    }
+    await fetch(`/campaign/${id}/steps`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json', 
+        'Authorization': `Bearer ${getJWT()}`
```

Comment:
Have we created a general util yet for POST and GET requests?

Reply:
I believe we have, but it's not name specific so I prefer to just use this for now until we do later

Reply:
Actually, that can be used, I'll try using it later and see.